channel/contractbackend.go

```diff
@@ -165,10 +187,7 @@ func (c *ContractBackend) nonce(ctx context.Context, sender common.Address) (uin
 		err = cherrors.CheckIsChainNotReachableError(err)
 		return 0, errors.WithMessage(err, "fetching nonce")
 	}
-
-	// Look up expected next nonce locally.
-	c.nonceMtx.Lock()
-	defer c.nonceMtx.Unlock()
+	SharedMutex.Lock()
```

Comment:
Since you also release the mutex directly when this function is ending, you can also just write 
`defer SharedMutex.Unlock()` after locking it and then remove the unlocking in line 203.